tests/test_supervision.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import pytest
+import unittest
+from datetime import timedelta, datetime
+from mock import Mock
+from protoactor.supervision import OneOfOneStrategy, SupervisorDirective, Supervisor
+from protoactor.pid import PID
+from protoactor.process_registry import ProcessRegistry
+from protoactor.restart_statistics import RestartStatistics
+from protoactor.process import LocalProcess
+from protoactor.mailbox import MailBox
+from protoactor.messages import ResumeMailbox, Restart, Stop
+
+@pytest.fixture(scope='module', )
+def supervisor_data():
+    supervisor = Supervisor()
+    mailbox = MailBox()
+    local_process = LocalProcess(self.mailbox)
+    pid_child = PID(address='address', id='id', aref=self.local_process)
+    restart_statistic = RestartStatistics(5, datetime(2017, 2, 15))
+
+    return {
+        'supervisor': supervisor,
+        'mailbox': mailbox,
+        'local_process': local_process,
+        'pid_child': pid_child,
+        'restart_statistic': restart_statistic
+    }
+
+
+    def test_handle_failure_resume_directive(supervisor_data):

```

Comment:
I think you need to indent out the test functions, because now it seems that they are under `supervisor_data` scope.